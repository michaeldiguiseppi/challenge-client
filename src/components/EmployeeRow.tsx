import React from "react";
import { Employee } from "./EmployeeList";
import "./EmployeeRow.css";

type Props = {
  employee: Employee;
};

const default_img_url = "/images/blank_profile_image_thumb.png";

const EmployeeRow: React.FC<Props> = ({ employee }) => {
  const { name, email, picture } = employee;
  return (
    <div data-testid="employee-row" className="employee-row">
      <div className="employee-row-top-container">
        <div className="employee-row-photo">
          <img
            src={picture?.thumbnail || default_img_url}
            alt={`${name?.first}-${name?.last}`}
          />
        </div>
        <h4>{`${name?.title}. ${name?.first} ${name?.last}`}</h4>
      </div>
      <div className="employee-row-bottom-container">
        <p>Email: {email}</p>
      </div>
    </div>
  );
};

export default EmployeeRow;
