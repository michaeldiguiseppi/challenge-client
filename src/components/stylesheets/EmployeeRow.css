.employee-row {
  border: 1px solid #8f8f8f;
  border-radius: 5px;
}

.employee-row-photo {
  padding: 15px;
}

.employee-row-photo img {
  display: flex;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 1px solid #8f8f8f;
}

.employee-row-top-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.employee-row-bottom-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}
