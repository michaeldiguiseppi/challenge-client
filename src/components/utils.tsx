import type { Employee } from "./types";

export const mockEmployees: Employee[] = [
  {
    name: {
      first: "Niclas",
      last: "Stornes",
      title: "Mr",
    },
    email: "niclas.stornes@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/78.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Erik",
      last: "Banks",
      title: "Mr",
    },
    email: "erik.banks@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/67.jpg",
      large: "https://randomuser.me/api/portraits/men/67.jpg",
      medium: "https://randomuser.me/api/portraits/med/men/67.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Andrea",
      last: "Crespo",
      title: "Mrs",
    },
    email: "andrea.crespo@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/women/41.jpg",
      large: "https://randomuser.me/api/portraits/women/41.jpg",
      medium: "https://randomuser.me/api/portraits/med/women/41.jpg",
    },
    id: null,
  },
];
