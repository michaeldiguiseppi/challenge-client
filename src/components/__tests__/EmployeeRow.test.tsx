import React from "react";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import EmployeeRow from "../EmployeeRow";
import { mockEmployees } from "../utils";

beforeEach(cleanup);

describe("<EmployeeRow />", () => {
  it("renders without crashing", () => {
    const { getByText, container } = render(
      <Router>
        <EmployeeRow employee={mockEmployees[1]} />
      </Router>
    );
    getByText(/erik banks/i);
    expect(container).toMatchSnapshot();
  });

  it("should render a thumbnail image for each employee", () => {
    const { getByAltText } = render(
      <Router>
        <EmployeeRow employee={mockEmployees[1]} />
      </Router>
    );
    getByAltText(/erik-banks/i);
  });

  it("should render a name for each employee", () => {
    const { getByText } = render(
      <Router>
        <EmployeeRow employee={mockEmployees[1]} />
      </Router>
    );
    getByText(/erik banks/i);
  });
});
