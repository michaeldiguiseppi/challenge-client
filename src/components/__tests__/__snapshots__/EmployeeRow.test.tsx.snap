// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployeeRow /> renders without crashing 1`] = `
<div>
  <a
    href="/people/erik.banks%40example.com"
  >
    <div
      class="employee-row"
      data-testid="employee-row"
    >
      <div
        class="employee-row-top-container"
      >
        <div
          class="employee-row-photo"
        >
          <img
            alt="Erik-Banks"
            src="https://randomuser.me/api/portraits/thumb/men/67.jpg"
          />
        </div>
        <div
          class="employee-row-name"
        >
          Mr. Erik Banks
        </div>
      </div>
    </div>
  </a>
</div>
`;
